support/cas-server-support-ehcache-ticket-registry/src/main/java/org/apereo/cas/config/EhcacheTicketRegistryConfiguration.java

```diff
@@ -159,13 +168,16 @@ public TicketRegistry ticketRegistry(@Qualifier("ehcacheTicketCacheManager") fin
                 LOGGER.debug("TicketCache.timeToIdle=[{}]", config.getTimeToIdleSeconds());
                 LOGGER.debug("TicketCache.cacheManager=[{}]", ehcache.getCacheManager().getName());
             }
-            manager.addDecoratedCacheIfAbsent(ehcache);
+            ehCacheManager.addDecoratedCacheIfAbsent(ehcache);
         });
 
+        manager.initializeCaches();
```

Comment:
This seems slightly problematic; is the manager going to skip caches that are already initialized? Without this change, where and how are the caches initialized?

Reply:
This is the Spring Ehcache manager which wraps the normal EhCacheManager and it is really just initializing an array of cache names. The initializedCaches() gets called in the afterpropertiesset() method of the Spring Ehcache manager but the caches aren't created when the ehcacheTicketCacheManager is created, they are only created when the ticketRegistry bean is created so it has to be called again after the caches are created, just to initialized the array (both cache managers have their own arrays of cache names).